go/internal/logservice/grpc/record_log_service.go

```diff
@@ -67,3 +78,25 @@ func (s *Server) PullLogs(ctx context.Context, req *logservicepb.PullLogsRequest
 	log.Info("PullLogs success", zap.String("collectionID", req.CollectionId), zap.Int("recordCount", len(records)))
 	return res, nil
 }
+
+func (s *Server) GetAllCollectionInfoToCompact(ctx context.Context, req *logservicepb.GetAllCollectionInfoToCompactRequest) (*logservicepb.GetAllCollectionInfoToCompactResponse, error) {
+	res := &logservicepb.GetAllCollectionInfoToCompactResponse{}
+	res.AllCollectionInfo = make([]*logservicepb.CollectionInfo, 0)
+	var recordLogs []*dbmodel.RecordLog
+	recordLogs, err := s.logService.GetAllCollectionIDsToCompact()
+	if err != nil {
+		log.Error("error getting collection info", zap.Error(err))
+		return nil, grpcutils.BuildInternalGrpcError("error getting collection info")
+	}
+	for index := range recordLogs {
```

Comment:
for _, recordLog := range recordLogs {
}

Reply:
sure